### PR TITLE
[STAT] update static analysis rules documentation

### PR DESCRIPTION
**Pull request by dastrong**
### What does this PR do?

This adds the `python-security` ruleset and updates the documentation for the other current rulesets. The related sourcing PR for these changes can be found [here](https://github.com/DataDog/datadog-static-analyzer-rule-docs/pull/10).

### Motivation

We want to release our Python security rules and update the current documentation as we've added new rules.

### Additional Notes

The sourcing PR linked above should be reviewed and approved prior to merging this.

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 1]
```diff
@@ -369,6 +490,28 @@ A set of rules to check that pandas code is used appropriately.
...
 
+### Ensure your Python code is safe and secure
+
+**Ruleset ID:** `python-security`
+
+# Python Security
+
+This ruleset is focused on analyzing code and finding any security issues in your code. It covers the OWASP10 and the SANS25 errors.
+
+It inspects Python code and finds security and vulnerabilities issues such as:
```
Comment (urseberry) [inline content/en/continuous_integration/static_analysis/rules/_index.md:501]:
```suggestion
It inspects Python code and finds security and vulnerability issues such as:
```

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 2]
```diff
@@ -271,6 +313,85 @@ python_pandas_data:
...
+    text: "use env vars over hardcoded values"
+  - link: "/continuous_integration/static_analysis/rules/python-security/deserialize-untrusted-data"
+    tag: "deserialize-untrusted-data"
+    text: "avoid unsafe function to (de)serialize data"
+  - link: "/continuous_integration/static_analysis/rules/python-security/file-write-others"
+    tag: "file-write-others"
+    text: "do not let all users write permissions"
+  - link: "/continuous_integration/static_analysis/rules/python-security/hardcoded-tmp-file"
+    tag: "hardcoded-tmp-file"
+    text: "Do not hardcoded tmp file or directory"
```
Comment (urseberry) [inline content/en/continuous_integration/static_analysis/rules/_index.md:337]:
```suggestion
    text: "Do not hardcode temp file or directory"
```

**Comment by dastrong** [comment]
@urseberry addressed your comments and made the python-security description more concise (and removed the H1 tag). Please re-review

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 3]
```diff
@@ -369,6 +490,24 @@ A set of rules to check that pandas code is used appropriately.
 
 <br>
 
+### Ensure your Python code is safe and secure
+
+**Ruleset ID:** `python-security`
+
+Rules focused on finding security and vulnerability issues in your Python code including those found in the OWASP10 and SANS25.
```
Comment (drichards-87) [inline content/en/continuous_integration/static_analysis/rules/_index.md:497]:
```suggestion
Rules focused on finding security and vulnerability issues in your Python code, including those found in the OWASP10 and SANS25.
```

**Pull request merged**